tensorflow/python/compiler/tensorrt/test/auto_mixed_precision_test.py

```diff
@@ -0,0 +1,129 @@
+                 dilation_rate=(1, 1)):
+  dtype = inputs.dtype
+  c_axis = -1 if data_format == "channels_last" else 1
+  nchan = inputs.shape[c_axis]
+  weights_shape = (kernel_size[0], kernel_size[1], nchan, filters)
+  weights = constant_op.constant(np.random.randn(*weights_shape), dtype=dtype)
+  padding = padding.upper()
+  if data_format == "channels_last":
+    strides = [1] + list(strides) + [1]
+    dilations = [1] + list(dilation_rate) + [1]
+    data_format = "NHWC"
+  else:
+    strides = [1, 1] + list(strides)
+    dilations = [1, 1] + list(dilation_rate)
+    data_format = "NCHW"
+  return gen_nn_ops.conv2d(
+      inputs,
+      weights,
+      strides=strides,
+      padding=padding,
+      dilations=dilations,
+      data_format=data_format)
+
+
+def build_graph(inp,
+                dtype,
+                num_filters,
+                data_format,
+                kernel_sizes,
+                dilation_rates,
+                padding="same"):
+  results = []
+  for kernel_size in kernel_sizes:
+    for dilation_rate in dilation_rates:
+      result = conv2d_layer(inp, num_filters, kernel_size, (1, 1), padding,
+                            data_format, dilation_rate)
+      results.append(result)
+  output = sum(results)
+  return array_ops.identity(output, name="output_0")
+
+
+class AutoMixedPrecisionTest(trt_test.TfTrtIntegrationTestBase):
+  """Testing TF-TRT conversion with `auto_mixed_precision` grappler pass."""
+
+  def GraphFn(self, inp):
+    np.random.seed(1234)
+    return build_graph(
+        inp=inp,
+        dtype=dtypes.float32,
+        num_filters=5,
+        data_format="channels_last",
+        kernel_sizes=[(3, 3), (3, 2)],
+        dilation_rates=[(1, 1), (2, 3)])
+
+  def GetParams(self):
+    # TODO(aaroey): test graph with different dtypes.
+    return self.BuildParams(self.GraphFn, dtypes.float32, [[13, 7, 11, 3]],
+                            [[13, 7, 11, 5]])
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
+    expected_engines = {
+        "TRTEngineOp_000": [
+            "Conv2D_3",
+            "Conv2D_2",
+            "Conv2D_1",
+            "Conv2D",
+            "add_1",
+            "add_2",
+            "add_3",
+        ]
+    }
+
+    if run_params.precision_mode != "FP32":
+      expected_engines["TRTEngineOp_000"].extend([
+          "Conv2D_3-1-CastToFp16-AutoMixedPrecision",
+          "Conv2D_2-1-CastToFp16-AutoMixedPrecision",
+          "Conv2D_1-1-CastToFp16-AutoMixedPrecision",
+          "Conv2D-1-CastToFp16-AutoMixedPrecision",
+          "Conv2D-0-CastToFp16-AutoMixedPrecision",
+          "add_3-1-CastToFp32-AutoMixedPrecision",
+          "add_2-1-CastToFp32-AutoMixedPrecision",
+          "add_1-1-CastToFp32-AutoMixedPrecision",
+          "add-1-CastToFp32-AutoMixedPrecision",
+      ])
+
+    return expected_engines
+
+  def ShouldRunTest(self, run_params):
+    """Whether to run the test."""
+    return True, ""
```

Comment:
Can you explain why we need to loosen the restriction defined [here](https://github.com/tensorflow/tensorflow/blob/1cbe90fbbf2b0670fda2a6b5b0764906e9bf0008/tensorflow/python/compiler/tensorrt/test/tf_trt_integration_test_base.py#L282-L286)? #

Reply:
Fair point, I'll use the base class method